lib/srv/usermgmt_linux.go

```diff
@@ -71,3 +75,31 @@ func (*HostUsersProvisioningBackend) DeleteUser(name string) error {
 	}
 	return trace.Wrap(err)
 }
+
+func (*HostUsersProvisioningBackend) TestSudoersFile(contents []byte) error {
+	code, err := host.TestSudoersFile(contents)
+	if err != nil {
+		if code != 0 {
+			return trace.Errorf("visudo: invalid sudoers file")
+		}
+		return trace.Wrap(err)
+	}
+	return nil
+}
+
+func (u *HostUsersProvisioningBackend) WriteSudoersFile(username string, contents []byte) error {
```

Comment:
I think `WriteSudoersFile` should call `TestSudoersFile` internally, and refuse to write an invalid file.